File: app/src/main/java/com/example/marijah/outflow/room_database/ExpenseDao.java
package com.example.marijah.outflow.room_database;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;

import java.util.ArrayList;
import java.util.List;

@Dao
public interface ExpenseDao
{
    @Query("select * from expenses")
    List<Expense> getExpenseList();

    @Insert
    void insertExpense(Expense expense);

    @Delete
    void deleteExpense(Expense expense);
}

File: app/src/main/java/com/example/marijah/outflow/room_database/ExpenseDao.kt
package com.example.marijah.outflow.room_database

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query

import java.util.ArrayList

@Dao
interface ExpenseDao {
    @get:Query("select * from expenses")
    val expenseList: List<Expense>

    @Insert
    fun insertExpense(expense: Expense)

    @Delete
    fun deleteExpense(expense: Expense)
}
